pkgs/development/interpreters/nextflow/default.nix

```diff
@@ -1,54 +1,102 @@
-{ lib
-, stdenv
-, fetchurl
-, makeWrapper
-, openjdk17
-, wget
-, which
-, gnused
-, gawk
-, coreutils
-, buildFHSEnv
+{
+  lib,
+  stdenv,
+  fetchurl,
+  makeWrapper,
+  openjdk,
+  gradle,
+  wget,
+  which,
+  gnused,
+  gawk,
+  coreutils,
+  testers,
+  nextflow,
+  bash,
 }:
-
-let
-  nextflow =
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: rec {
   pname = "nextflow";
-  version = "22.10.6";
+  version = "24.08.0-edge";
 
   src = fetchurl {
-    url = "https://github.com/nextflow-io/nextflow/releases/download/v${version}/nextflow-${version}-all";
-    hash = "sha256-zeYsKxWRnzr0W6CD+yjoAXwCN/AbN5P4HhH1oftnrjY=";
+    url = "https://github.com/nextflow-io/nextflow/archive/6e866ae81ff3bf8a9729e9dbaa9dd89afcb81a4b.tar.gz";
+    hash = "sha256-j1HFHn8Vu1/HLC1aAruFV7wm3otze8fTegAcf+cLrY4=";
   };
 
   nativeBuildInputs = [
     makeWrapper
-    openjdk17
+    gradle
+    bash
+    openjdk
     wget
     which
     gnused
     gawk
     coreutils
   ];
 
-  dontUnpack = true;
+  mitmCache = gradle.fetchDeps {
+    inherit (finalAttrs) pname;
+    data = ./deps.json;
+  };
+  __darwinAllowLocalNetworking = true;
+
+  patches = [ ./add-build-dependencies.patch ];
+
+  postPatch = ''
+    # Ironically, replacing 'bin/bash' by '/usr/bin/env [-S]' would work when
+    # actually using `nextflow` (in NixOS). However, that invocation fails
+    # within any Nix build environment (because there is no /usr/...). Thus,
+    # invoking `nextflow` within nixpkgs would be impossible.
+    substituteInPlace modules/nextflow/src/main/groovy/nextflow/executor/BashWrapperBuilder.groovy \
+      --replace-fail "['/bin/bash'," "['${bash}/bin/bash',"
+  '';
+
+  # See Makefile (`make pack`)
+  preBuild = ''
+    export BUILD_PACK=1
+  '';
+  gradleBuildTask = "pack";
 
   installPhase = ''
     runHook preInstall
 
     mkdir -p $out/bin
-    install -Dm755 $src $out/bin/nextflow
+    install -Dm755 build/releases/nextflow-${version}-dist $out/bin/nextflow
 
     runHook postInstall
   '';
 
   postFixup = ''
     wrapProgram $out/bin/nextflow \
       --prefix PATH : ${lib.makeBinPath nativeBuildInputs} \
```

Comment:
Sort of related: I've been cleaning up `error: makeWrapper/makeShellWrapper must be in nativeBuildInputs` in cross-compilation scenarios, and while `makeWrapper` is correctly placed in `nativeBuildInputs` here, using `lib.makeBinPath` on it causes the error to appear in builds of reverse dependencies (in this case `nf-test`). I think it would make sense to keep only `makeWrapper` inside the actual `nativeBuildInputs` attribute and move the rest (or at least the ones that are necessary) to be just passed directly to `lib.makeBinPath`. Currently still running a build without your PR, but with this change.

Reply:
(See this [commit](https://github.com/NixOS/nixpkgs/pull/341532/commits/b2b275a4ca9711351879a5d6b2db9aebd9530366) for an implementation of the suggested approach)

Reply:
@SFrijters I adapted the code as you suggested and sucessfully run `nix build -L path:.#nextflow.tests.hello`. I have updated this PR now.